### PR TITLE
Create a separate environment for testing schematic refactor branch 

### PR DESCRIPTION
**Pull request by linglp**
PR Checklist:
[ ] Describe and explain your intentions for this change
[ ] Setup pre-commit and run the validators (info in README.md)

Related to: https://sagebionetworks.jira.com/browse/FDS-1233

We need to create a separate schematic environment for testing schematic refactor. Ideally, the environment should be like "dev" environment that is under D&T dev account

### CODE REVIEW

File: .github/workflows/aws-deploy.yml [hunk 1]
```diff
@@ -90,6 +90,37 @@ jobs:
           cdk_subcommand: 'deploy'
           cdk_args: '--require-approval never --context env=staging --progress events --debug'
           actions_comment: false
+
+  cdk-deploy-dev-refactor:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/dev' }}
```
Comment (zaro0508) [inline .github/workflows/aws-deploy.yml]: shouldn't this execute based on a different branch?

File: .github/workflows/aws-deploy.yml [hunk 2]
```diff
@@ -90,6 +90,37 @@ jobs:
...
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::631692904429:role/sagebase-github-oidc-sage-bionetworks-schematic-infra
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: 1200
+
+      - name: cdk deploy to org-sagebase-dnt-dev (631692904429)
+        uses: youyo/aws-cdk-github-actions@v2
+        with:
+          cdk_subcommand: 'deploy'
+          cdk_args: '--require-approval never --context env=dev --progress events --debug'
```
Comment (zaro0508) [inline .github/workflows/aws-deploy.yml]: `env` should probably reference `dev-refactor`?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
i believe [secrets need to be setup for each environment](https://github.com/Sage-Bionetworks/schematic-infra#secrets), have you setup secrets yet?

File: .github/workflows/aws-deploy.yml (hunk 1 above)
Comment (linglp) [reply .github/workflows/aws-deploy.yml]: Yes. Thank you for the comment @zaro0508 !

File: .github/workflows/aws-deploy.yml (hunk 2 above)
Comment (linglp) [reply .github/workflows/aws-deploy.yml]: Yes. Good catch. Thank you

**Comment by linglp** [comment]
> i believe [secrets need to be setup for each environment](https://github.com/Sage-Bionetworks/schematic-infra#secrets), have you setup secrets yet?

Can I name my secret as: `schematic-dev-refactor-DockerFargateStack/dev-refactor/ecs`. Based on the documentation, the naming convention is: `<cfn stack id>/<environment id>/<secret name>`. Since my stack name prefix is: `schematic-dev-refactor`, so I assume that I should use `schematic-dev-refactor`.. but based on the screen shot of the failed deployment, it seems like the stack name is: `schematic/dev-refactor-dev-refactor-DockerFargateStack`? 
Screen Shot 2023-11-03 at 9 47 17 AM

What is the environment id? Should it just be `dev` or `dev-refactor`?

File: cdk.json [hunk 3]
```diff
@@ -41,6 +41,17 @@
...
+    "dev-refactor": {
+      "IMAGE_PATH_AND_TAG": "ghcr.io/sage-bionetworks/schematic:v23.9.3",
+      "AWS_DEFAULT_REGION": "us-east-1",
+      "PORT": "443",
+      "TAGS": {
+        "CostCenter": "NO PROGRAM / 000000"
+      },
+      "STACK_NAME_PREFIX": "schematic-dev-refactor",
+      "ACM_CERT_ARN": "arn:aws:acm:us-east-1:631692904429:certificate/0e9682f6-3ffa-46fb-9671-b6349f5164d6",
+      "VPC_CIDR": "10.255.73.0/24"
```
Comment (zaro0508) [inline cdk.json]: this CIDR is already used by `dev` env.  i suggest you pick a CIDR value that has not been used yet.  maybe "10.255.76.0/24"? more info about it here https://sagebionetworks.jira.com/wiki/spaces/IT/pages/2850586648/Setup+AWS+VPC

**Comment by zaro0508** [comment]
please take a look at the code that defines the <cfn stack id> , https://github.com/Sage-Bionetworks/schematic-infra/blob/dev/docker_fargate/docker_fargate_stack.py#L47

it's a bit of a concatenation of the stack prefix, the context and `DockerFargateStack`

File: cdk.json (hunk 3 above)
Comment (linglp) [reply cdk.json]: Okay thanks for pointing this out. I will update this part.

**Comment by linglp** [comment]
> please take a look at the code that defines the , https://github.com/Sage-Bionetworks/schematic-infra/blob/dev/docker_fargate/docker_fargate_stack.py#L47
> 
> it's a bit of a concatenation of the stack prefix, the context and `DockerFargateStack`

Thanks Khai. I think maybe the secret is set up correctly? I could see that the secret got retrieved on Nov.5th..

Screen Shot 2023-11-06 at 3 40 53 PM

**Comment by linglp** [comment]
@zaro0508 I noticed that the github action failed because the CIDR value was still referencing the wrong one after I updated it. After I deleted the old stack and redeployed again, it seems to work. 


Old stack: 
Screen Shot 2023-11-06 at 4 03 08 PM

New stack: 

Screen Shot 2023-11-06 at 4 06 22 PM

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**